src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -2213,7 +2213,20 @@ private BoundExpression BindOutVariableArgument(DeclarationExpressionSyntax decl
 
                 if (isVar)
                 {
-                    return new OutVariablePendingInference(declarationExpression, localSymbol, null);
+                    if (localSymbol.DeclarationKind == LocalDeclarationKind.OutVariable)
```

Comment:
>if (localSymbol.DeclarationKind == LocalDeclarationKind.OutVariable) [](start = 20, length = 68)

I believe that this method should never be called if this condition is not true. The fix should be applied elsewhere.